public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -267,17 +267,17 @@ export default class PromQlLanguageProvider extends LanguageProvider {
 
     // Stitch all query lines together to support multi-line queries
     let queryOffset;
-    const queryText = value.document.getBlocks().reduce((text: string, block) => {
+    const queryText = value.document.getBlocks().reduce((text, block) => {
       if (!block) {
-        return text;
+        return text!;
```

Comment:
hmmm... this reduce is a bit funny. I can't figure out why typescript thinks that text is potentially undefined, and i cannot reproduce this in a playground 😅

I don't think using `!` is great, as it's another way to just ignore typescript's type checking (however funny it might be right here), and I think we want to avoid these smells so they don't proliferate.

```typescript
    const queryText = value.document.getBlocks().reduce((text, block) => {
      if (!block || !text) {
        return text ?? '';
      }
```

This will do the job, and I think keep the functionality the same (though I would check with Observability that it's correct) - if text is undefined, just return an empty string and carry on

Reply:
Ahhhh - `getBlocks()` actually returns an `Immutable.List`, not an `Array`, with its own implementation/typing of reduce which seems to be incorrect.

either way - either 

```typescript
if (!text) {
  return ""
}
```

or the more compact (but slightly weirder) in the previous comment should do the job.

Reply:
ahhhh nice, yeah i couldn't work out why it was complaining either, i just put it down to a typescript bug 😂  nice job!